lib/common/http2_client.go

```diff
@@ -11,12 +12,16 @@ import (
 	"golang.org/x/net/http2"
 )
 
+type HTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
 type HTTP2Client struct {
-	client    http.Client
+	client    HTTPClient
```

Comment:
I feel that `HTTPClient` in `HTTP2Client` is awful. :->  How about `Doer` or `HTTPDoer`?